californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -219,9 +219,8 @@ public void receiveRequest(final Request request, final EndpointReceiver receive
 			duplicate = endpointContextMatcher.isToBeSent(previousSourceContext, sourceContext);
 			if (!duplicate) {
 				// the new context doesn't match the previous.
-				exchangeStore.remove(idByMID, previous);
-				if (exchangeStore.findPrevious(idByMID, exchange) != null) {
-					LOGGER.warn("new request could not be registered!");
+				if (!exchangeStore.replacePrevious(idByMID, previous, exchange)) {
+					LOGGER.warn("new request {} could not be registered!", request);
```

Comment:
could we add a log about the successful replacement  with old and new request ?

Reply:
OK, I'll add it.